streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -567,6 +575,13 @@ void suspend(final boolean clean,
                 commit(false);
             } finally {
                 if (eosEnabled) {
+                    // In #commit(), we don't write a checkpoint file when EOS is turned on.
```

Comment:
This logic is clumsy here: we can instead just remove the `if (!eosEnabled) {` check in `commit` and hence do not need this here any more.